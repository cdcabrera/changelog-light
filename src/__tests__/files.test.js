const { updateChangelog, updatePackage } = require('../files');
const { parseCommits } = require('../parse');

describe('Files', () => {
  it('should create, and update a CHANGELOG.md', () => {
    const commitLog = `
      1f12345b597123453031234555b6d25574ccacee refactor(file): lorem updates (#8)
      53a12345479ef91123456e921234548ac4123450 feat(dolor): issues/20 sit enhancements (#8)
      d1234537b5e94a6512345xeb96503312345x18d2 fix(build): eslint, jsdoc updates (#16)
      e5c456ea12345vv4610fa4aff7812345ss31b1e2 chore(build): npm packages (#15)
    `;

    const commitObj = parseCommits({ getGit: () => commitLog });

    expect(updateChangelog(commitObj, undefined, { date: '2022-10-01' })).toMatchSnapshot('changelog');
  });

  it('should update a package.json', () => {
    expect(updatePackage('lorem ipsum')).toMatchSnapshot('package');
  });
});
