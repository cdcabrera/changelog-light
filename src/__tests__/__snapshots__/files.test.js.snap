// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateChangelog should create a CHANGELOG.md basic 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

## 1.0.0 (2022-10-01)


### Features
* **dolor** issues/20 sit enhancements (#8) (53a1234)

### Code Refactoring
* **file** lorem updates (#8) (1f12345)

### Chores
* **build** npm packages (#15) (e5c456e)

### Bug Fixes
* **build** eslint, jsdoc updates (#16) (d123453)
*  missing semicolon  (1f1x345)

"
`;

exports[`updateChangelog should create a CHANGELOG.md with breaking changes 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

## 1.0.0 (2022-10-01)
⚠ BREAKING CHANGES

### Features
* ⚠ **lorem** lorem dolor sit (#12) (1f12345)
* **dolor** issues/20 sit enhancements (#8) (53a1234)

### Code Refactoring
* **file** lorem updates (#8) (1f12345)

### Chores
* **build** npm packages (#15) (e5c456e)

### Bug Fixes
* **build** eslint, jsdoc updates (#16) (d123453)
*  missing semicolon  (1f1x345)

"
`;

exports[`updateChangelog should create a CHANGELOG.md with breaking changes and description 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

## 1.0.0 (2022-10-01)
⚠ BREAKING CHANGES

Lorem ipsum dolor sit!

### Features
* ⚠ **lorem** lorem dolor sit (#12) (1f12345)
* **dolor** issues/20 sit enhancements (#8) (53a1234)

### Code Refactoring
* **file** lorem updates (#8) (1f12345)

### Chores
* **build** npm packages (#15) (e5c456e)

### Bug Fixes
* **build** eslint, jsdoc updates (#16) (d123453)
*  missing semicolon  (1f1x345)

"
`;

exports[`updateChangelog should create a CHANGELOG.md with no release commit 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

## 1.0.0 (2022-10-01)


### Features
* **dolor** issues/20 sit enhancements (#8) (53a1234)

### Code Refactoring
* **file** lorem updates (#8) (LASTg45)

### Chores
* **build** npm packages (#15) (e5c456e)

### Bug Fixes
* **build** eslint, jsdoc updates (#16) (d123453)
*  missing semicolon  (1f1x345)

"
`;

exports[`updateChangelog should create a CHANGELOG.md with release commit 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

## [1.0.0](https://localhost/lorem/ipsum/comparmock/REALFIRST2345d421231231312312345dca11235...REALFIRST2345d421231231312312345dca11235) (2022-10-01)


### Features
* **dolor** issues/20 sit enhancements (#8) (53a1234)

### Code Refactoring
* **file** lorem updates (#8) (LASTg45)

### Chores
* **release** 0.1.0  (REALFIR)
* **build** npm packages (#15) (e5c456e)

### Bug Fixes
* **build** eslint, jsdoc updates (#16) (d123453)
*  missing semicolon  (1f1x345)

"
`;

exports[`updateChangelog should create a CHANGELOG.md with release commit, description, and no version comparison markdown link 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

## 1.0.0 (2022-10-01)
### ⚠ **lorem ipsum**
- \`dolor\` (#1234)
- \`sit\` (#5678)

### Features
* **dolor** issues/20 sit enhancements (#8) (53a1234)

### Code Refactoring
* **file** lorem updates (#8) (LASTg45)

### Chores
* **release** 0.1.0  (REALFIR)
* **build** npm packages (#15) (e5c456e)

### Bug Fixes
* **build** eslint, jsdoc updates (#16) (d123453)
*  missing semicolon  (1f1x345)

"
`;

exports[`updatePackage should update a package.json: package 1`] = `"Version bump: lorem ipsum"`;
